pkg/manager/manager.go

```diff
@@ -66,11 +98,23 @@ func NewManager(config *config.Config) (*Manager, error) {
 			return nil, err
 		}
 		newManager.workers = append(newManager.workers, w)
-		newManager.workersLastInvokeTime = append(newManager.workersLastInvokeTime, time.Now().AddDate(-1, 0, 0))
+		name, _ := repoConfig["name"].(string)
+		if _, ok := newManager.workersLastInvokeTime[name]; !ok {
+			newManager.workersLastInvokeTime[name] = time.Now().AddDate(-1, 0, 0)
+		}
 	}
 	return &newManager, nil
 }
 
+func (m *Manager) checkpoint() error {
+	file, _ := json.MarshalIndent(m.workersLastInvokeTime, "", "  ")
+	err := ioutil.WriteFile(m.config.Checkpoint, file, 0644)
```

Comment:
perhaps we should use atomic file write here like https://pkg.go.dev/github.com/siddontang/go/ioutil2#WriteFileAtomic (though that implementation missed a f.sync() before f.close()).